### DISCUSSION TITLE
v3.9.0 Goals

### DISCUSSION THREAD

**Discussion by bevans2000** [mars-sim/mars-sim#1342] [General]
What are the goals of 3.9.0?

- Further implement the Project based Mission framework which allows a generic and flexible means to create missions; https://github.com/orgs/mars-sim/projects/4

**Comment by bevans2000** [comment]
@mokun What else do you want to add?

**Reply by mokun** [reply]
I definitely want to improve the height resolution. 

With more fine grain height resolution, we'll be able to make missions travel NOT in straight lines anymore, as height resolution is associated with whether a vehicle may move forward or turn to the left or the right.

**Reply by bevans2000** [reply]
> I definitely want to improve the height resolution. 
> 
> With that, I'd like to make missions travel NOT in straight lines anymore, as height resolution is associated with when a vehicle turns to the left or the right.
> 
> 
> 

We can visit this as part of our Mission rework

**Reply by mokun** [reply]
@bevans2000 

So far, we push a total of 46 *open* bugs and features into the v3.9.0 bucket : 
https://github.com/mars-sim/mars-sim/milestone/16

Let's rack and stack !

**Reply by mokun** [reply]
The thing is that some of these issues are initially and partially being worked on a few milestones ago.

The GitHub issue tracker doesn't have that ability to set up multiple milestones for each issue.

**Reply by bevans2000** [reply]
We need to lock down a couple of high level objectives. I have some changes ready for the Loading/Unloading change on a branch 

What about the Unit hierarchy changes? That would help solve the concurrent access issues and the teleporting

**Reply by mokun** [reply]
I just created a new temporary placeholder milestone called 3.x.x and relocate a whole bunch of open issues that were assigned to 3.9.0 to go there.

Now, [3.9.0](https://github.com/mars-sim/mars-sim/milestones/3.9.0) only has 13 opens, while [3.x.x](https://github.com/mars-sim/mars-sim/milestone/19) has 34 opens.

**Reply by mokun** [reply]
@bevans2000 @muybien3433 

After we're done with optimizing the mineral layer, can we work on [this](https://github.com/mars-sim/mars-sim/issues/959) ?

It's totally unrealistic to drive on a straight line from a settlement to a site.

There might be bugs associated with how an angle or a direction is selected when driving. 

It's supposed to be sensitive to the elevation changes.

If it's fixed, at least we can see a zigzagging line on a mission.

**Reply by bevans2000** [reply]
Ok we can look at it. I want to do the improvements to the map window first.
